Tools/CMake/AMReXThirdPartyLibraries.cmake

```diff
@@ -6,9 +6,29 @@ if (AMReX_HDF5)
        set(HDF5_PREFER_PARALLEL TRUE)
     endif ()
     find_package(HDF5 1.10.4 REQUIRED)
```

Comment:
Purely informational:

If HDF5 was build with CMake and not with its legacy (but still widely used) autotools installer, then `find_package` can use the CMake config package that it installed (based on a hint to the root path) to pick up HDF5's install config. This installation method would not be affected and is the one that the Spack package uses already.

In order to use this, we would pass `CONFIG` here as an argument (or require CMake 3.24+, more below). This is what the response here https://github.com/HDFGroup/hdf5/issues/2488#issuecomment-1442303115 refers to.

For the default `find_package` signature that we currently use, and prior to CMake 3.24 which we also currently use, the `FindHDF5.cmake` module takes precedence before checking for the Config package.
https://cmake.org/cmake/help/latest/command/find_package.html#search-modes
https://cmake.org/cmake/help/latest/command/find_package.html#search-procedure